modules/reference/appvms/comms.nix

```diff
@@ -69,12 +69,33 @@ in
       # GPSD collects data from GPS and makes it available on TCP port 2947
       services.gpsd = {
         enable = true;
+        # Give the ttyUSB0 device for gpsd in case gps was plugged in from boot
+        # This doesn't affect anything if device is not available
         devices = [ "/dev/ttyUSB0" ];
+        # Use safer read-only mode
         readonly = true;
-        debugLevel = 2;
+        # Set debug level to zero so gpsd won't flood the logs unnecessarily
+        debugLevel = 0;
+        # Listen on all IP-addresses
         listenany = true;
-        extraArgs = [ "-n" ]; # Do not wait for a client to connect before polling
+        # Do not wait for a client to connect before polling
+        nowait = true;
+        # Give gpsd the control socket to use, so it will keep running even if there are no gps devices connected
+        extraArgs = [
+          "-F"
+          "/var/run/gpsd.sock"
+        ];
       };
+      # When USB gps device is inserted run gpsdctl to add the device to gpsd, so it starts monitoring it
+      # (Note that this will be run way before gpsd service is running, if device is already connected when booting.
+      # This does not seem to have any negative effects though)
+      services.udev.extraRules = ''
+        ACTION=="add", ENV{ID_BUS}=="usb", ENV{ID_VENDOR_ID}=="067b", ENV{ID_MODEL_ID}=="23a3", ENV{DEVNAME}=="/dev/ttyUSB*", RUN+="${pkgs.gpsd}/bin/gpsdctl add '%E{DEVNAME}'"
```

Comment:
Nice fix. Do you think it would make sense to get the USB VID and PID from `gps0` in `config.ghaf.hardware.usb.external` to avoid hardcoding it in multiple places?

Reply:
Good idea, implemented.